feat: premium servers list search bar

dpaulos6 opened the pull request.

dpaulos6 committed:
feat: premium servers list search bar

dpaulos6 committed:
feat: code improvements

tobias-hay commented on app/premium/page.tsx:
```diff
@@ -296,6 +315,8 @@ export default function Premium() {
 										href="https://stripe.com/"
 										target="_blank"
 										className="flex flex-wrap justify-center gap-2 sm:gap-3"
+										rel="noreferrer noopener"
```
This has been the default behavior for quite some time now, so we don't need to explicitly write this

[Chrome](https://issues.chromium.org/issues/40599708)
[Safari](https://bugs.webkit.org/show_bug.cgi?id=155166)
[FireFox](https://bugzilla.mozilla.org/show_bug.cgi?id=1222516)

```suggestion

```

tobias-hay reviewed (changes requested):
Looks good, though this will need to be re-written when #104 gets merged since this file gets changed in that PR.

dpaulos6 committed:
feat: remove `rel="noreferrer noopener"`

Co-authored-by: Woofer <61364477+Woofer21@users.noreply.github.com>

dpaulos6 commented:
> Looks good, though this will need to be re-written when #104 gets merged since this file gets changed in that PR.

Thanks for letting me know! Will keep this in mind and make sure to rewrite as soon as #104 gets merged.

mezotv commented:
Just merged the other pr @dpaulos6 feel free to request review here again once possible conflicts are gone!

dpaulos6 committed:
Merge branch 'main' into feature/searchbar-premium-servers-select

dpaulos6 committed:
feat: adjust to match PR #104 + formatting

dpaulos6 committed:
chore(style): format files

dpaulos6 committed:
feat: remove unnecessary imports

dpaulos6 committed:
Merge branch 'feature/searchbar-premium-servers-select' of https://github.com/Would-You-Bot/website into feature/searchbar-premium-servers-select

dpaulos6 committed:
feat: minor fixes

dpaulos6 committed:
chore(style): format files

dpaulos6 committed:
Delete .hintrc

dpaulos6 committed:
Merge branch 'feature/searchbar-premium-servers-select' of https://github.com/Would-You-Bot/website into feature/searchbar-premium-servers-select

tobias-hay commented on components/Premium/Content.tsx:
```diff
@@ -105,131 +104,146 @@ export default function Premium() {
+											</DialogHeader>
+											<DialogDescription
+												className="w-full !rounded-2xl"
+												asChild
```
This is not a prop that DialogDescription has, no need for it.
```suggestion

```

tobias-hay commented on components/Premium/Content.tsx:
```diff
@@ -105,131 +104,146 @@ export default function Premium() {
+												</DialogTitle>
+											</DialogHeader>
+											<DialogDescription
+												className="w-full !rounded-2xl"
```
Neither of these classes do anything, they are not needed. 
```suggestion

```

tobias-hay commented on components/Premium/Content.tsx:
```diff
@@ -105,131 +104,146 @@ export default function Premium() {
+												className="w-full !rounded-2xl"
+												asChild
+											>
+												<Fragment>
```
This fragment is not necessary with the "asChild" removed"
```suggestion

```

tobias-hay commented on components/Premium/Content.tsx:
```diff
@@ -249,63 +263,68 @@ export default function Premium() {
-											:	null}
-										</Dialog>
-									</div>
+												</Fragment>
```
Closing fragment not needed with the opening fragment gone
```suggestion

```

tobias-hay commented on components/Premium/Content.tsx:
```diff
@@ -20,24 +11,32 @@ import {
 import { CheckIcon, Link } from 'lucide-react'
 import { useIdToken } from '@/helpers/hooks'
-import { Suspense, useState } from 'react'
+import { Fragment, useState } from 'react'
```
We can remove this Fragment import based on removals below.
```suggestion
import { useState } from 'react'
```

tobias-hay reviewed (changes requested):
In addition to the below changes, it would also be nice to get a loading state for the <Command> while the server list is loading.

dpaulos6 committed:
Merge branch 'main' into feature/searchbar-premium-servers-select

dpaulos6 committed:
feat: removing unnecessary code

dpaulos6 committed:
chore(style): format files

tobias-hay reviewed (changes requested):
Pretty good, 2 things:

1. Page crashes when you try to navigate to it while logged out
2. Lets added a loading state & disable the button after you press it & its waiting for the stripe page to open

tobias-hay committed:
fix: no server -> no servers

woofer21 committed:
chore(style): format files

dpaulos6 committed:
feat: skeleton loading & servers api endpoint

Also added JSDoc description to redis and ratelimit functions aswell as a bit of code improvements.

dpaulos6 committed:
feat: disable on checkout+ improvements

tobias-hay commented on app/premium/_components/checkoutButton.tsx:
```diff
@@ -1,30 +1,38 @@
+import { Fragment, useEffect, useState } from 'react'
 import { loadStripe } from '@stripe/stripe-js'
-import { useState } from 'react'
+import { LoadingSvg } from '@/icons/loading'
```
Lets replace this with the lucide icon, looks better imo and matches the rest of the apps icons.
```suggestion
import { LoaderCircle } from 'lucide-react'
```

tobias-hay commented on app/premium/_components/checkoutButton.tsx:
```diff
@@ -56,48 +64,24 @@ export default function CheckoutButton({
-					</svg>
+			{isLoading ?
+				<Fragment>
+					<LoadingSvg />
```
Update it here as well.
```suggestion
					<LoaderCircle className="animate-spin mr-2" />
```

tobias-hay commented on components/Premium/Content.tsx:
```diff
@@ -30,10 +30,12 @@ import PlansComparison from '@/app/premium/_components/PlansComparison'
-import { useEffect, useState } from 'react'
-import { getServer } from '@/lib/redis'
+import { LoadingSvg } from '@/icons/loading'
+import { Button } from '../ui/button'
```
Not used, can be removed
```suggestion
```

tobias-hay commented on icons/loading.tsx:
```diff
@@ -0,0 +1,34 @@
+			</clipPath>
+		</defs>
+	</svg>
+)
```
Since its not used anywhere else (assuming you do the above suggestions) we can delete this.
```suggestion
```

tobias-hay reviewed (changes requested):
Few changes to loading state, otherwise looking good.

dpaulos6 committed:
feat: updated loading state

Co-Authored-By: Woofer <61364477+Woofer21@users.noreply.github.com>

mezotv commented:
@dpaulos6 @Woofer21 were the issues fixed?

tobias-hay commented:
> @dpaulos6 @Woofer21 were the issues fixed?

Should be good

dpaulos6 commented:
> @dpaulos6 @Woofer21 were the issues fixed?

i guess so, i can run a second check, will update soon.

mezotv commented:
> i guess so, i can run a second check, will update soon.

Alright let me know once done so I can merge!

mezotv commented:
@Woofer21 just approve ig.

dpaulos6 commented:
Yea it seems that i address all of the requests. LGTM!

tobias-hay reviewed (approved):
lgtm
